File: backend/urbfix-api/src/main/java/com/urbfix/api/models/Chamado.java
package com.urbfix.api.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "chamado")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Chamado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cidadao_id")
    private Cidadao cidadao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id")
    private CategoriaProblema categoria;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column(name = "score_prioridade")
    private Double scorePrioridade;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoChamado estado;

    @Column(name = "imagem_base64", columnDefinition = "TEXT")
    private String imagemBase64;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: mobile/urbfix_mobile/lib/models/chamado.dart
class Chamado {
  final int id;
  final int? cidadaoId;
  final String? cidadaoNome;
  final int categoriaId;
  final String? categoriaNome;
  final double? pesoCategoria;
  final double latitude;
  final double longitude;
  final double scorePrioridade;
  final String estado;
  final String createdAt;

  const Chamado({
    required this.id,
    this.cidadaoId,
    this.cidadaoNome,
    required this.categoriaId,
    this.categoriaNome,
    this.pesoCategoria,
    required this.latitude,
    required this.longitude,
    required this.scorePrioridade,
    required this.estado,
    required this.createdAt,
  });

  factory Chamado.fromJson(Map<String, dynamic> json) {
    return Chamado(
      id: json['id'] as int,
      cidadaoId: json['cidadaoId'] as int?,
      cidadaoNome: json['cidadaoNome'] as String?,
      categoriaId: json['categoriaId'] as int,
      categoriaNome: json['categoriaNome'] as String?,
      pesoCategoria: (json['pesoCategoria'] as num?)?.toDouble(),
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      scorePrioridade: (json['scorePrioridade'] as num).toDouble(),
      estado: json['estado'] as String,
      createdAt: json['createdAt'] as String,
    );
  }
}
